src/test/scala/com/mongodb/spark/NoSparkConfSpec.scala

```diff
@@ -59,9 +61,28 @@ class NoSparkConfSpec extends RequiresMongoDB {
     ds.collect().toList should equal(characters)
   }
 
+  it should "use the passed in connector" in {
+    val writeConfig = WriteConfig(Map("uri" -> mongoClientURI, "database" -> databaseName, "collection" -> collectionName))
+    val documents = sc.parallelize((1 to 10).map(i => Document.parse(s"{test: $i}")))
+    documents.saveToMongoDB(writeConfig = writeConfig)
+
+    val connector = MongoConnector(CustomMongoClientFactory(mongoClientURI))
+    val readConfig = ReadConfig(Map("uri" -> "mongodb://example.com/", "database" -> databaseName, "collection" -> collectionName,
+      "partitioner" -> "TestPartitioner$"))
+    val mongoSpark = MongoSpark.builder().sparkContext(sc).connector(connector).readConfig(readConfig).build()
+
+    mongoSpark.toRDD().collect().size should equal(10)
+    mongoSpark.toDF().count() should equal(10)
```

Comment:
Without the change this fails.

Reply:
Because it uses the ReadConfig with "mongodb://example.com" URI?  If so... ack!